### PR TITLE
Run Task SDK tests without DB

### PR DESCRIPTION
**Pull request by kaxil**
We don't need Database for Task SDK. So let's skip initializing or using the database. This will help us catch if there is a code that needs DB leaks in the SDK code

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by ashb — COMMENTED** [review]
Hmmm, could this be done in the task_sdk/tests/conftest.py too somehow, so that if you run `pytest` it will also not spin up the DB?

I think there is a marker we could apply globally via that file.

**Comment by kaxil** [comment]
> Hmmm, could this be done in the task_sdk/tests/conftest.py too somehow, so that if you run `pytest` it will also not spin up the DB?
> 
> I think there is a marker we could apply globally via that file.

@ashb Done

**Comment by kaxil** [comment]
Waiting for https://github.com/apache/airflow/pull/43185 to get merged first

**Review by Lee-W — APPROVED** [review]

File: task_sdk/tests/conftest.py [hunk 1]
```diff
@@ -16,10 +16,15 @@
...
 from __future__ import annotations
 
+import os
+
 import pytest
 
 pytest_plugins = "tests_common.pytest_plugin"
 
+# Task SDK does not need access to the Airflow database
+os.environ["_AIRFLOW_SKIP_DB_TESTS"] = "true"
```
Comment (pankajkoti) [inline task_sdk/tests/conftest.py:26]: I believe we’re running the Task SDK tests in a separate process, as indicated by the dedicated task-sdk-tests.yml GitHub workflow. However, I hope that we aren’t running any other tests/modules in the same process elsewhere that could be impacted by this setup.

**Review by pankajkoti — APPROVED** [review]

File: task_sdk/tests/conftest.py (hunk 1 above)
Comment (ashb) [reply task_sdk/tests/conftest.py:26]: Pytest won't let you run yet unrelated test folders - i.e. ones that don't share a conftest.py in parent dir, and some there isn't a one at the top level of the repo we're good

**Pull request merged**